crates/chia-consensus/src/spendbundle_conditions.rs

```diff
@@ -108,6 +116,36 @@ mod tests {
             .fold(0, |sum, spend| sum + spend.create_coin.len());
         assert_eq!(create_coins, additions);
         assert_eq!(conditions.cost, cost);
+        // Generate a block with the same spend bundle and compare its cost
+        let program_spends: Vec<_> = bundle
+            .coin_spends
+            .iter()
+            .map(|coin_spend| {
+                (
+                    coin_spend.coin,
+                    &coin_spend.puzzle_reveal,
+                    &coin_spend.solution,
+                )
+            })
+            .collect();
+        let program = solution_generator(program_spends).expect("solution_generator failed");
+        let blocks: &[&[u8]] = &[];
+        let block_conds = run_block_generator2::<_, MempoolVisitor, _>(
+            &mut a,
+            program.as_slice(),
+            blocks,
+            11_000_000_000,
+            MEMPOOL_MODE,
+            &TEST_CONSTANTS,
+        )
+        .expect("run_block_generator2 failed");
+        // The QUOTE_COST difference here is because get_conditions_from_spendbundle
+        // uses solution_generator which wraps the program in a quote, and
+        // while that quote's byte cost is included in the cost calculation, its
+        // execution cost is not, because get_conditions_from_spendbundle doesn't
+        // run that generator.
```

Comment:
The cost difference comes from the overhead of creating and executing a block generator (i.e. `solution_generator()` + `run_block_generator2()`) versus just the cost of the spend bundles. The spend bundles don't pay the size-cost nor execution cost of being wrapped by a quote.